packages/zosjobs/src/SearchJobs.ts

```diff
@@ -0,0 +1,162 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { AbstractSession, ImperativeError, IRestClientResponse, IOptionsFullResponse, Headers, Logger} from "@zowe/imperative";
+import { ZosmfRestClient, ZosmfHeaders } from "@zowe/core-for-zowe-sdk";
+import { GetJobs} from "./GetJobs";
+import { ISearchJobsParms } from "./doc/input/ISearchJobsParms";
+import { IJobFile } from "./doc/response/IJobFile";
+import { IJob } from "./doc/response/IJob";
+import { JobsConstants } from "./JobsConstants";
+import { ZosJobsMessages } from "./JobsMessages";
+
+/**
+ * Class to handle the searching of z/OS batch job spool output
+ * @export
+ * @class SearchJobs
+ */
+export class SearchJobs {
+    /**
+     * Search Jobs - Search the spool output of the specified job name for a string or regular expression
+     * @static
+     * @param {AbstractSession} session - z/OSMF connection info
+     * @param {ISearchJobsParms} searchParms - The search parameters for the API call
+     * @returns {Promise<string>} - promise that resolves to spool output
+     * @throws {ImperativeError} --search-string or --search-regx must be specified
+     * @memberof searchJobs
+     */
+    public static async searchJobs(session: AbstractSession, searchParms : ISearchJobsParms) {
+        const jobName: string = searchParms.jobName;
+        const searchString: string= searchParms.searchString;
+        const searchRegex: string = searchParms.searchRegex;
+        const caseInsensitive: boolean = searchParms.caseInsensitive;
+        const searchLimit: number = searchParms.searchLimit;
+        const fileLimit: number = searchParms.fileLimit;
+
+        Logger.getAppLogger().info("SearchJobs.searchJobs() called!");
+        let replyBuffer:string = "";
+
+        // Validate that a search string or regex parameter was passed
+        if(searchRegex === undefined && searchString === undefined){
+            throw new ImperativeError({ msg: ZosJobsMessages.missingSearchOption.message });
+        }
+
+        // Validate that both options are not passed on the same call
+        if(searchRegex !== undefined && searchString !== undefined){
+            throw new ImperativeError({ msg: ZosJobsMessages.missingSearchOption.message });
+        }
+
+        const jobsList: IJob[] = await GetJobs.getJobsByPrefix(session, jobName);
+        let fileCount = 0;
+        for(const job of jobsList )
```

Comment:
Not saying that these weren't tested, since there are system, integration, and unit tests (for the handler and definition files)

But I have to agree with Codecov on this one. 
We should probably add one or two unit-level API tests to make sure that these lines are covered.

Reply:
I can write a API test, but it isn't going to do much, as so much of this code is dependent on responses from zosmf. I can see I can mock them up. We do not have them for ViewJob either.

Reply:
I do agree that the tests won't be doing much (compared to the rests of the tests that you already have written in this PR) 😋 

We should probably discuss how we can consider these scenarios in the future (i.e. the code has been thoroughly tested outside of the SDK/API unit-level testing

Reply:
I've added new unit and system test. The code coverage requirements for the API should now pass.